scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -260,6 +260,22 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
                             "and $scannerCriteria, not scanning the package again $packageIndex."
                 }
 
+                val packagePath = pkg.id.toPath()
+                val unarchivedProvenances = HashSet<Provenance>()
+
+                storedResults.forEach {
+                    if (!archiver.hasArchive("$packagePath/${it.provenance.hash()}")) {
+                        unarchivedProvenances.add(it.provenance)
+                    }
```

Comment:
Did you check how long this takes for a large scan with several thousand dependencies? If it is more than a few seconds consider wrapping the loop with `measureTime` and adding a performance log with `log.perf`.

Reply:
Thanks for checking, 30s doesn't sound like much, but if you run scans in a CI/CD environment it can be relevant. Usually you only need this check if you changed the location of the artifact storage or if you use scan results from an external provider, so this check is not always required. I'm just thinking if it would be better to add a command line option to the scanner to enable/disable this check. What do you think @sschuberth @fviernau ?

Reply:
Yes, let's add a scanner option for this that is exposed as a CLI flag and defaults to `false`. I propose a flag name of `--create-missing-archives`.